src/Compilers/CSharp/Portable/Symbols/Tuples/TupleFieldSymbol.cs

```diff
@@ -0,0 +1,233 @@
+        {
+            get
+            {
+                return true;
+            }
+        }
+
+        /// <summary>
+        /// If this field represents a tuple element (including the name match), 
+        /// id is an index of the element (zero-based).
+        /// Otherwise, (-1 - [index in members array]);
+        /// </summary>
+        public int TupleFieldId
+        {
+            get
+            {
+                return _tupleFieldId;
+            }
+        }
+
+        public override FieldSymbol TupleUnderlyingField
+        {
+            get
+            {
+                return _underlyingField;
+            }
+        }
+
+        public override Symbol AssociatedSymbol
+        {
+            get
+            {
+                Symbol underlyingAssociated = _underlyingField.AssociatedSymbol;
+                if ((object)underlyingAssociated == null)
+                {
+                    return null;
+                }
+
+                return _containingTuple.GetTupleMemberSymbolForUnderlyingMember(underlyingAssociated);
+            }
+        }
+
+        public override Symbol ContainingSymbol
+        {
+            get
+            {
+                return _containingTuple;
+            }
+        }
+
+        public override ImmutableArray<CustomModifier> CustomModifiers
+        {
+            get
+            {
+                return _underlyingField.CustomModifiers;
+            }
+        }
+
+        internal override TypeSymbol GetFieldType(ConsList<FieldSymbol> fieldsBeingBound)
+        {
+            return _underlyingField.GetFieldType(fieldsBeingBound);
+        }
+
+        public override ImmutableArray<CSharpAttributeData> GetAttributes()
+        {
+            return _underlyingField.GetAttributes();
+        }
+
+        internal override DiagnosticInfo GetUseSiteDiagnostic()
+        {
+            DiagnosticInfo result = base.GetUseSiteDiagnostic();
+            MergeUseSiteDiagnostics(ref result, _underlyingField.GetUseSiteDiagnostic());
+            return result;
+        }
+
+        public override sealed int GetHashCode()
+        {
+            return Hash.Combine(_containingTuple.GetHashCode(), _tupleFieldId.GetHashCode());
+        }
+
+        public override sealed bool Equals(object obj)
+        {
+            return Equals(obj as TupleFieldSymbol);
+        }
+
+        public bool Equals(TupleFieldSymbol other)
+        {
+            if ((object)other == this)
+            {
+                return true;
+            }
+
+            return (object)other != null && _tupleFieldId == other._tupleFieldId && _containingTuple == other._containingTuple;
+        }
+    }
+
+    /// <summary>
+    /// Represents an element field of a tuple type (such as (int, byte).Item1)
+    /// that is backed by a real field with the same name within the tuple underlying type.
+    /// </summary>
+    internal class TupleElementFieldSymbol : TupleFieldSymbol
```

Comment:
No, it is one of TupleRenamedElementFieldSymbol.

---
In reply to: [61812537](https://github.com/dotnet/roslyn/pull/11006#discussion_r61812537) [](ancestors = 61812537)